### ISSUE TITLE
 Exception: Class '_LocalClassMirror' has no instance getter 'getters'.

### ISSUE THREAD

**Issue by bitnick10** [chrisbu/dartwatch-JsonObject#40]
I'm newbie to dart.I wonder to know is this a bug or my fault.
I'm using dart 1.12.1,dartium 1.11.3.0 and Intellij IDEA
When I run the sample

class Other { String name = "My Name"; }

class Basic {
  String myString = "foo";
  int myInt = 42;
  Other name = new Other();
}

void main() {
  var basic = new Basic();
  objectToJson(basic).then((jsonString) => print(jsonString));
}

I got the error:

Exception: Class '_LocalClassMirror' has no instance getter 'getters'.

NoSuchMethodError: method not found: 'getters'
Receiver: Instance of '_LocalClassMirror'
Arguments: []package:json_object/src/mirror_based_serializer.dart:127 _serializeObject@1003131927package:json_object/src/mirror_based_serializer.dart:53 objectToSerializablepackage:json_object/src/mirror_based_serializer.dart:22 objectToJsonindex.dart:56 main

**Comment by stevenroose** [comment]
I personally have no knowledge over the `objectToJson` method. @chrisbu wrote that one, so hopefully he can help.

As far as I can understand from the code, the code seems to be made for an older version of `dart:mirrors` that had a `ClassMirror.getters` property. The pubspec only mentions a minimum sdk version, so maximum, so probably a refactor has happened when around 1.0.

I have no interest in rewriting the code for the mirror-based code. In fact, now that it's broken, it might be better to remove it, that would also benefit JS compile size.

**Comment by thosakwe** [comment]
Yeah, I had this same error, so I went through the code. A lot of the mirror-based code in this repository is aimed for older versions of Dart. I didn't want to break the library for current users by changing the underlying code, so I just made my own library that does a similar thing.

**So basically, this issue is fixed with [thosakwe/json_god](https://github.com/thosakwe/json_god).** The API is the tiniest bit different, though. Rather than using a JSONObject class, you can serialize/deserialize any Dart type, using the newer mirror API. Examples are in the repo README.md. _It works in Dart2JS, too._ Just include a `@MirrorsUsed()` annotation on your import to 'dart:mirrors'.

I've used it in a few smaller projects, and I think it does the job well for SDK ^1.0.0.

**Comment by stevenroose** [comment]
@thosakwe Off topic, but I've got [this awesome Chrome extension](https://chrome.google.com/webstore/detail/nicolas-cage/fjgbnnhgejbfgfbofeohlffjlpceejlm?hl=en):

![](https://i.imgur.com/l7N71Bd.png)

**Comment by thosakwe** [comment]
Haha, that's funny.
